lib/elixir/lib/enum.ex

```diff
@@ -1419,17 +1424,22 @@ defmodule Enum do
   If multiple elements are considered minimal, the first one that was found
   is returned.
 
-  Raises `Enum.EmptyError` if `enumerable` is empty.
+  Calls the provided `empty_fallback` and returns its value if `enumerable`
+  is empty. The default `empty_fallback` raises `Enum.EmptyError`.
 
   ## Examples
 
       iex> Enum.min([1, 2, 3])
       1
 
+      iex> Enum.min([], fn -> 0 end)
+      0
+
   """
-  @spec min(t) :: element | no_return
-  def min(enumerable) do
-    reduce(enumerable, &Kernel.min(&1, &2))
+  @spec min(t, (() -> empty_result)) :: element | empty_result | no_return when empty_result: any
+  def min(enumerable, empty_fallback \\ fn -> raise Enum.EmptyError end)
+  def min(enumerable, empty_fallback) when is_function(empty_fallback, 0) do
```

Comment:
This is a bit nitpicky, sorry, but we usually leave an empty line between a bodyless clause and its implementation 😄